src/software/embedded/robot_localizer.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <Eigen/Dense>
+#include <deque>
+#include "software/geom/angle.h"
+#include "software/geom/angular_velocity.h"
+#include "software/sensor_fusion/filter/kalman_filter.h"
+#include "proto/world.pb.h"
+#include "time.h"
+#include "software/embedded/services/imu.h"
+
+
+class RobotLocalizer {
```

Comment:
did you use a resource to implement this? it'd be nice to link it for future reference